src/v/datalake/tests/schema_registry_test.cc

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "bytes/iobuf.h"
+#include "datalake/schema_registry.h"
+#include "model/record.h"
+#include "utils/vint.h"
+
+#include <gtest/gtest.h>
+
+#include <variant>
+
+TEST(DatalakeSchemaRegistry, SchemaIdForValidRecord) {
+    iobuf value;
+
+    uint8_t magic = 0;
+    int32_t schema_id = 12;
+    int32_t schema_id_encoded = htobe32(schema_id);
+    std::string payload = "Hello world";
+    value.append(&magic, 1);
+    value.append(reinterpret_cast<uint8_t*>(&schema_id_encoded), 4);
+    value.append(payload.data(), payload.size());
+
+    auto res = datalake::get_value_schema_id(value);
+    ASSERT_TRUE(res.has_value());
+    EXPECT_EQ(res.value().schema_id(), schema_id);
+    EXPECT_EQ(res.value().shared_message_data.size_bytes(), payload.size());
+}
+
+TEST(DatalakeSchemaRegistry, SchemaIdForShortRecord) {
+    iobuf value;
+
+    uint8_t magic = 0;
+    int32_t schema_id = 12;
+    int32_t schema_id_encoded = htobe32(schema_id);
```

Comment:
TBH I find the name `htobe32` to be quite horrible (not your fault). Seastar has nice helpers for this: `ss::cpu_to_be`

Reply:
Done.